lib/measured/unit.rb

```diff
@@ -13,13 +13,13 @@ def initialize(name, aliases: [], value: nil)
   def name_eql?(name_to_compare, case_sensitive: false)
     return false unless name_to_compare.present?
     name_to_compare = name_to_compare.to_s
-    case_sensitive ? @name.eql?(name_to_compare) : case_insensitive(@name).include?(name_to_compare.downcase)
+    case_sensitive ? @name.eql?(name_to_compare) : (@name.casecmp(name_to_compare) == 0)
   end
 
   def names_include?(name_to_compare, case_sensitive: false)
     return false unless name_to_compare.present?
     name_to_compare = name_to_compare.to_s
-    case_sensitive ? @names.include?(name_to_compare) : case_insensitive(@names).include?(name_to_compare.downcase)
+    case_sensitive ? @names.include?(name_to_compare) : downcased_names.include?(name_to_compare.downcase)
```

Comment:
Lazily created and cache a downcased names list to avoid having to create it every time.